tests/spam_test.py

```diff
@@ -22,3 +22,32 @@ def test_spam_configuration(self):
         self.testbot.assertCommand('!plugin config SpammingAlert '
                                    '{\'MAX_MSG_LEN\':1, \'MAX_LINES\': 1}',
                                    'configuration done')
+
+
+class TestSpamExtraConfig(unittest.TestCase):
+
+    def setUp(self):
+        extra_config = {
+            'DEFAULT_CONFIG': {
+                'SpammingAlert': {
+                    'MAX_MSG_LEN': 500,
+                    'MAX_LINES': 5,
+                }
+            }
+        }
+        self.testbot = TestBot(extra_plugin_dir='plugins',
+                               loglevel=logging.ERROR,
+                               extra_config=extra_config)
+        self.testbot.start()
+
+    def test_spam_extra_config_callback(self):
+        self.testbot.assertCommand('c'*5001, 'you\'re spamming')
```

Comment:
this does not confirm that the extra config is being used, because 5001 is above the default of `1000` and the extra config value of `500`.  So it will only verify that the command works; it doesnt verify that the command is triggered only if above the number of lines setting.

An obvious omission from this test module is asserting that a person can post content below these sizes and no errors occur.  I'm not sure whether errbot test framework has a suitable helper for that.

the next line is also not 100% ok.   `'\n\n'*6` is 12 new line characters, and that is lower than `constants.MAX_LINES` -- so, all good?

not quite ... it assumes that pytest did not run `test_spam_extra_config_configuration` before `test_spam_extra_config_callback`.
There is no guarantee of that.
https://duckduckgo.com/?q=pytest+plugin+random+order

Tests should not leave the test bed modified after they have finished

Simplest fix is to add a `tearDown` if using `unittest.TestCase` to reset the config to the default state, or ... less correct .. (re)set to a constant state in `setUp`.

This also needs to be done to `TestSpam`.

Reply:
Shit. That 5001 was a typo. It was supposed to be 501. :sweat_smile: